### ISSUE TITLE
Automatically set header in `prepareJsonPayload`

### ISSUE THREAD

**Issue by mskocik** [php-mod/curl#100]
I would suggest to automatically set header `Content-type: application/json` when calling `prepareJsonPayload`. Typically when doing post as json. 
What do you think? I could prepare PR for this change.

**Comment by nadar** [comment]
Hi @mskocik 

I totally agree, but this would introduce backwards compatibility break. So we need (even its ugly) another parameter for doing this, like $contenTypeJson which is false by default.

With this in mind, yes please send a PR :+1: Thanks

**Comment by nadar** [comment]
We are now in 3.0 dev branch, i will implement a solution for this, maybe a new method.

**Comment by mskocik** [comment]
I agree with new method. Because although I think it's right to set headers in `prepareJsonPayload` it isn't that flexible. That's also the reason why I haven't pushed to PR.
